### ISSUE TITLE
[Documentation] How to customize (or what to customize) for a sign out call to Azure AD B2C?

### ISSUE THREAD

**Issue by henriquebelotto** [AzureAD/microsoft-identity-web#1112]
### Documentation related to component
Sign Out -> OnRedirectToIdentityProviderForSignOut method

### Please check all that apply

- [ ] typo
- [x] documentation doesn't exist
- [x] documentation needs clarification
- [ ] error(s) in the example
- [x] needs an example

### Description of the issue
Hello,
In our application (.net core 3.1 MVC) we sign in users using Azure AD B2C using Custom policies with the Microsoft Identity Web library (this library) and we need to different policies to Sign in, one for regular users (signInSignUp) and one for admin only (signInAdmin).
I was able to create different "path" and authenticate the user using the required policy basically by overriding the "accountController" from this library and using the following methods:

```csharp
  public IActionResult SignIn()
        {
            // Audit log for signIn is done in AzureADB2COpenIdConnectOptionsConfigurator class
            string scheme = OpenIdConnectDefaults.AuthenticationScheme;
            //string scheme = "AzureAdB2C";
            var redirectUrl = Url.Content("~/");

            var test = Challenge(
                new AuthenticationProperties { RedirectUri = redirectUrl },
                scheme);

            return Challenge(
                new AuthenticationProperties { RedirectUri = redirectUrl },
                scheme);
        }

        public IActionResult SignInAdmin()
        {
            //string scheme = "AzureAdB2CAdmin";
            string scheme = OpenIdConnectDefaults.AuthenticationScheme;

            var redirectUrl = Url.Content("~/");
            var properties = new AuthenticationProperties { RedirectUri = redirectUrl };
            properties.Items[Constants.Policy] = "B2C_1A_SignInAdmin";
            return Challenge(properties, scheme);
        }
```

It works fine to sign in the user for the correct policy.
However, my issue is how to sign out based on the policy?

I tried to override the Sign Out method as following:
```CSharp
[HttpGet("{scheme?}")]
        public IActionResult SignOut([FromRoute] string scheme)
        {
                var claimsIdentity = (ClaimsIdentity)HttpContext.User.Identity;

                var properties = new AuthenticationProperties { RedirectUri = "/Account/SignOut" };
               
               // Fake claim just to tell if it's an Admin or a regular user that is authenticated
                if (claimsIdentity.HasClaim(c => c.Type == "Test"))
                {
                    properties.Items[Constants.Policy] = "B2C_1A_SignInAdmin";
                }

                string auditMessage = HttpDetail.GetAuditLog(Request.HttpContext);
                _logger.CreateAuditLog(auditMessage);

                scheme ??= OpenIdConnectDefaults.AuthenticationScheme;
                //var callbackUrl = Url.Page("", pageHandler: null, values: null, protocol: Request.Scheme);
                return SignOut(
                     properties,
                     CookieAuthenticationDefaults.AuthenticationScheme,
                     scheme);
            }         

        }
```

And the same way that I did for Sign In, by setting the policy, here
` properties.Items[Constants.Policy] = "B2C_1A_SignInAdmin";`

But it doesn't seem to work and I'm still sending the user to the "main" sign in policy, which is the SignInSignUp.

So, my question is what I need to change to send the user to sign out from the correct policy (or sign out from all the policies at the same that, that would be okay as well)?

**Comment by henriquebelotto** [comment]
I was able to achieve what I'm trying by modifying the IssuerAddress in the ProtocolMessage, inside the method 
`OnRedirectToIdentityProviderForSignOut(RedirectContext context)`

`context.ProtocolMessage.IssuerAddress = context.ProtocolMessage.IssuerAddress.Replace("b2c_1a_signuporsignin", "b2c_1a_signinadmin");`

It seems to work fine now. But is it correct what I did?
Is there a better way to do it?

**Comment by jmprieur** [comment]
@henriquebelotto I think that this is fine.
you might want to do that by chaining the OpenIdConnect event. See https://github.com/AzureAD/microsoft-identity-web/wiki/customization#openidconnectoptions

**Comment by TiagoBrenck** [comment]
@jmprieur we actually do it for the event `OnRedirectToIdentityProvider` (https://github.com/AzureAD/microsoft-identity-web/blob/master/src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs#L365) 
The above code has a comment explaining why this interceptor is needed. I ve noticed that there is another event called `OnRedirectToIdentityProviderForSignOut`. It might worth testing if this event also needs the same B2C handler.

**Comment by jmprieur** [comment]
Good point, @TiagoBrenck !

**Comment by jennyf19** [comment]
@henriquebelotto Fixed in 1.18.0 release. closing.
